common/lexer/flex_lexer_adapter.h

```diff
@@ -133,6 +149,9 @@ class FlexLexerAdapter : private CodeStreamHolder, protected L, public Lexer {
 
   // Contains the enumeration and the substring slice of the last lexed token.
   TokenInfo last_token_;
+
+  // Used to track characters discarded from the end of the token text
+  size_t discarded_ = 0;
```

Comment:
This is when I wish we already had benchmarks to measure the performance impact of these changes.
I suspect that `UpdateLocation()` is one of the hottest functions, given that every lexer rule calls it.
Think about this, I hesitate to bloat `FlexLexerAdapter` for every subclass, including the `VerilogLexer`.

Is there a way we could perform token edits like `SkipPrefix` and `DiscardSuffix` outside of this lexer code, perhaps in a separate transformative pass over the token sequence?

Alternatively, lex the unwanted prefix and suffix as their own tokens that can be discarded from parsing by filtering.

Reply:
Removed the extra callbacks and added some pre-processing right after `MakeTokenSequence`